### PR TITLE
Bump up version to 1.0.4 for adding startTls support

### PR DESCRIPTION
**Pull request by cifer828**
## Description

Add startTls support in version 1.0.4

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Major release (change is NOT backward compatible with prior release)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ This library is built and managed using [maven](https://maven.apache.org/what-is
 <dependency>
   <groupId>com.yahoo.smtpnio</groupId>
   <artifactId>smtpnio.core</artifactId>
-  <version>1.0.3</version>
+  <version>1.0.4</version>
 </dependency>
 ```
```
Comment (fansu) [inline README.md:49]: can you check line 84 blocks and add sample usage and explanation on how to use setEnableStartTls(true)

Comment (cifer828) [reply README.md:49]: sure

Comment (fansu) [reply README.md:49]: Give a short description of what your starttls will do, such as you will do ssl first, fallback logic, ehlo etc.

File: README.md [hunk 2]
```diff
@@ -145,12 +147,28 @@ smtpClient.shutdown();
...
+This release, version 1.0.4, supports the following SMTP commands:
 - **EHLO**
 - **HELO**
 - **AUTH** (PLAIN, LOGIN, XOAUTH2)
 - **QUIT**
 
+Version 1.0.4 adds startTls support.
+- **Both ssl and startTls options are enabled**
+
+     Client will first try normal ssl and detect the connection result. If successul, secure session will be created. If failed, client will close the old connection and create a new plain one. Then try startTls as fallback.
```
Comment (fansu) [inline README.md:159]: fix typo, successul

Comment (cifer828) [reply README.md:159]: sure

**Comment by fansu** [comment]
Looks ok.

**Pull request merged**